app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java

```diff
@@ -305,22 +306,32 @@ private void updateCurrent() {
                             if (!statuses.isEmpty()) {
                                 filterStatuses(statuses);
 
-                                // Working around a bug when Mastodon API doesn't return the first
-                                // status because of string "id < maxId". Hacking with ID doesn't
-                                // help.
-                                if (!this.statuses.isEmpty()) {
-                                    Either<Placeholder, Status> firstOld = this.statuses.get(0);
-                                    this.statuses.clear();
```

Comment:
So this line basically was the problem: it removes all statuses. But there might have been a concurrent top load going on in the meantime that inserted newer stauses. I fixed it by clearing only statuses that are older than the `topId` at the moment the load was started.

Reply:
Is the mastodon bug mentioned in the comment still addressed now?

Reply:
I am pretty sure it is, but I will check again

Reply:
We should consider using new API parameters finally but not in this PR ofc